cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -476,48 +490,87 @@ bool UnorderedRespHelper<algorithmFPType, cpu>::findBestSplitCategoricalFeature(
+        }
     }
 }
 
 template <typename algorithmFPType, CpuType cpu>
-int UnorderedRespHelper<algorithmFPType, cpu>::findBestSplitForFeatureSorted(algorithmFPType * featureBuf, IndexType iFeature, const IndexType * aIdx,
-                                                                             size_t n, size_t nMinSplitPart, const ImpurityData & curImpurity,
-                                                                             TSplitData & split, const algorithmFPType minWeightLeaf,
-                                                                             const algorithmFPType totalWeights) const
+void UnorderedRespHelper<algorithmFPType, cpu>::computeHistFewClassesWithWeights(IndexType iFeature, const IndexType * aIdx, size_t n) const
 {
-    const auto nDiffFeatMax = this->indexedFeatures().numIndices(iFeature);
-    _idxFeatureBuf.setValues(nDiffFeatMax, algorithmFPType(0));
-    _weightsFeatureBuf.setValues(nDiffFeatMax, algorithmFPType(0));
-    _samplesPerClassBuf.setValues(nClasses() * nDiffFeatMax, 0);
+    const IndexedFeatures::IndexType * const indexedFeature = this->indexedFeatures().data(iFeature);
+    const auto aResponse                                    = this->_aResponse.get();
+    const auto aWeights                                     = this->_aWeights.get();
+
     auto nFeatIdx         = _idxFeatureBuf.get();
-    auto featWeights      = _weightsFeatureBuf.get();
     auto nSamplesPerClass = _samplesPerClassBuf.get();
 
-    countResponses<algorithmFPType, typename super::Response, IndexType, typename IndexedFeatures::IndexType, size_t, cpu>(
-        _nClasses, n, aIdx, this->_aResponse.get(), this->indexedFeatures().data(iFeature), nFeatIdx, nSamplesPerClass, featWeights,
-        this->_aWeights.get());
+    {
+        for (size_t i = 0; i < n; ++i)
+        {
+            const IndexType iSample              = aIdx[i];
+            const auto & r                       = aResponse[aIdx[i]];
+            const IndexedFeatures::IndexType idx = indexedFeature[r.idx];
+            ++nFeatIdx[idx];
+            const auto weights          = aWeights[iSample].val;
+            const ClassIndexType iClass = r.val;
+            nSamplesPerClass[idx * _nClasses + iClass] += weights;
+        }
+    }
+}
+
+template <typename algorithmFPType, CpuType cpu>
+void UnorderedRespHelper<algorithmFPType, cpu>::computeHistManyClasses(IndexType iFeature, const IndexType * aIdx, size_t n) const
+{
+    const IndexedFeatures::IndexType * const indexedFeature = this->indexedFeatures().data(iFeature);
+    const auto aResponse                                    = this->_aResponse.get();
+    const auto aWeights                                     = this->_aWeights.get();
+
+    auto nFeatIdx         = _idxFeatureBuf.get();
+    auto featWeights      = _weightsFeatureBuf.get();
+    auto nSamplesPerClass = _samplesPerClassBuf.get();
 
     algorithmFPType bestImpDecrease =
```

Comment:
remove it, have a look original [PR](https://github.com/oneapi-src/oneDAL/pull/923) (please check that library was successfully built locally before push update)

Reply:
Corrected